### DISCUSSION TITLE
The project was not built due to "Failed to init ct.sym for .."

### DISCUSSION THREAD

**Discussion by hvbtup** [eclipse-birt/birt#713] [Q&A] [answered]
I think this is not a BIRT specific issue, but related to the Eclipse IDE in general.

As I already mentioned in https://github.com/eclipse/birt/discussions/712, when I use the Eclipse installer to setup a BIRT development environment, the build fails with errors.

I downloaded jdk-11.0.13+8 (Temurin) and the generated eclipse.ini contains

```
-vm
C:\prog\jdk\jdk-11.0.13+8\bin

```

But nevertheless an older Java 11 AdoptOpenJDK is used by the installed Eclipse for building:

![grafik](https://user-images.githubusercontent.com/7644776/141969201-dd0dc3e6-ade2-4182-aaaa-b19572b20499.png)

And the build fails miserably. One of the error messages is 
```
The project was not built due to "Failed to init ct.sym for C:\prog\jdk\jdk-11.0.2+9-jre\lib\jrt-fs.jar"
```

In order to build BIRT (with Eclipse), I have to explicitly select the other "Installed JRE".

I _guess_ this is because the 11.0.2+9-jre is only a JRE, not a full JDK.

But the error message is _so_ useless. I didn't find any information what ct.sym is (a file?) and what purpose it has.
That's why I wanted to mention it here.

After this (and a Clean/Build), most of the errors disappeared, but the birt-runtime-test project still showed errors.

I had to uncheck "enable project specific settings" here:
![grafik](https://user-images.githubusercontent.com/7644776/141971861-10cd95c3-66ea-496c-9970-f5410d2291a0.png)

And now it looks ok:
![grafik](https://user-images.githubusercontent.com/7644776/141972118-01adfe95-e793-4cdf-990a-d9684f588a33.png)

**Comment by ruspl-afed** [comment]
@hvbtup  What if you will uncheck "Use `--release` option"?

Here is the brief info what the `ct.sym` is https://www.morling.dev/blog/the-anatomy-of-ct-sym-how-javac-ensures-backwards-compatibility/

**Reply by hvbtup** [reply]
I didn't test that, maybe it would help.
I don't like to change settings I don't understand.
Disabling project specific settings worked for me. Can this cause problems and, if yes, more or less than unchecking the --release option?

**Reply by wimjongman** [reply]
Henning, thanks for opening a new discussion.

I also noticed that the birt-runtime-test was already converted to java11 (see issue #672)

> Can this cause problems and, if yes, more or less than unchecking the --release option?

Yes, this can cause problems when people have different settings. e.g. code formatting, on the workspace level. 

The birth charts test bundle has this as a preference. Can you set the preference like that and see if that solves the issue?


![image](https://user-images.githubusercontent.com/180969/141986238-e70d539c-45d9-4a59-8c65-4f4c81fa9f2d.png)

**Reply by wimjongman** [reply]
With "like that" I mean checking that checkbox?

**Reply by hvbtup** [reply]
This seems to help. Not quite sure, because I tried different other things.

**Reply by wimjongman** [reply]
You can go to the project and "Replace with/HEAD Revision" then apply this change again.

**Comment by hvbtup** [comment, marked as answer]
As I wrote: I guess this is because the 11.0.2+9-jre I used was only a _JRE_, not a full JDK.
